packages/botonic-nlp/src/tasks/text-classification/process/processor.ts

```diff
@@ -0,0 +1,53 @@
+import { tensor } from '@tensorflow/tfjs-node'
+
+import { Sample } from '../../../dataset/types'
+import { LabelEncoder } from '../../../encode/label-encoder'
+import { OneHotEncoder } from '../../../encode/one-hot-encoder'
+import { PADDING_TOKEN, UNKNOWN_TOKEN } from '../../../preprocess/constants'
+import { Preprocessor } from '../../../preprocess/preprocessor'
+import { InputData, OutputData } from './types'
+
+export class Processor {
+  constructor(
+    readonly preprocessor: Preprocessor,
+    readonly tokensEncoder: LabelEncoder,
+    readonly classEncoder: OneHotEncoder
+  ) {}
+
+  process(samples: Sample[]): { x: InputData; y: OutputData } {
```

Comment:
rename process to processSamples and generateInput to processTexts (to be consistent with processText). Consistency above all! :-)

Reply:
agree :)